consensus/spos/bls/subroundBlock.go

```diff
@@ -140,11 +156,32 @@ func (sr *subroundBlock) sendBlock(header data.HeaderHandler, body data.BodyHand
 		return false
 	}
 
+	var signatureShare []byte
+	if sr.EnableEpochsHandler().IsFlagEnabled(common.ConsensusPropagationChangesFlag) {
+		selfIndex, err := sr.ConsensusGroupIndex(leader)
+		if err != nil {
+			log.Debug("sendBlock.SelfConsensusGroupIndex: leader not in consensus group")
+			return false
+		}
+
+		headerHash := sr.Hasher().Compute(string(marshalizedHeader))
+		signatureShare, err = sr.SigningHandler().CreateSignatureShareForPublicKey(
```

Comment:
you could also save the signature share for the leader, to be used afterwards in aggregation.